monitorcontrol/monitorcontrol.py

```diff
@@ -135,6 +159,160 @@ def _get_vcp_feature(self, code: vcp.VCPCode) -> int:
         current, maximum = self.vcp.get_vcp_feature(code.value)
         return current
 
+    def _extract_a_cap(self, caps_str: str, key: str) -> str:
+        """
+        Splits the capabilities string into individual sets.
+
+        Returns:
+            Dict of all values for the capability
+        """
+        start_of_filter = caps_str.upper().find(key.upper())
+
+        if start_of_filter == -1:
+            # not all keys are returned by monitor.
+            # Also sometimes the string has errors.
+            return ""
+
+        start_of_filter += len(key)
+        filtered_caps_str = caps_str[start_of_filter:]
+        end_of_filter = 0
+        for i in range(len(filtered_caps_str)):
+            if filtered_caps_str[i] == "(":
+                end_of_filter += 1
+            if filtered_caps_str[i] == ")":
+                end_of_filter -= 1
+            if end_of_filter == 0:
+                # dont change end_of_filter to remove the closing ")"
+                break
+
+        # 1:i to remove the first character "("
+        return filtered_caps_str[1:i]
+
+    def _convert_to_dict(self, caps_str: str) -> dict:
+        """
+        Parses the VCP capabilities string to a dictionary.
+        Non continuous capabilities will include an array of
+        all supported values.
+
+        Returns:
+            Dict with all capabilities in hex
+
+        Example:
+            Expected string "04 14(05 06) 16" is converted to
+            {
+                0x04: [],
+                0x14: [0x05, 0x06],
+                0x16: [],
+            }
+        """
+
+        if len(caps_str) == 0:
+            # Sometimes the keys arent found and the extracting of
+            # capabilities returns an empty string.
+            return {}
+
+        start = 0
+        result_dict = {}
+        sub_key = None
+
+        for i in range(len(caps_str)):
+            if caps_str[i] in " ":
+                val = caps_str[start:i]
+
+                # sub arrays can end in ") " which is detected as
+                # an empty value. Filter those out.
+                if val == "":
+                    continue
+
+                val = int(val, 16)
+                if sub_key is None:
+                    result_dict[val] = []
+                else:
+                    result_dict[sub_key].append(val)
+
+                start = i + 1  # +1 to skip the space
+            if caps_str[i] in "(":
+                sub_key = int(caps_str[start:i], 16)
+                result_dict[sub_key] = []
+                start = i + 1  # +1 to skip the (
+            if caps_str[i] in ")":
+                val = caps_str[start:i]
+
+                if val == "":
+                    continue
+
+                val = int(val, 16)
+                result_dict[sub_key].append(val)
+
+                sub_key = None
+                start = i + 1  # +1 to skip the )
+        return result_dict
+
+    def _parse_capabilities(self, caps_str: str) -> dict:
```

Comment:
This does use `_extract_a_cap`, but that can be removed from the class so this can too.